lang/java/avro/src/test/java/org/apache/avro/TestUuidConversions.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.avro;
+
+import org.apache.avro.generic.GenericFixed;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.UUID;
+import java.util.stream.Stream;
+
+public class TestUuidConversions {
+
+  private Conversions.UUIDConversion uuidConversion = new Conversions.UUIDConversion();
+
+  private Schema fixed = Schema.createFixed("fixed", "doc", "", Long.BYTES * 2);
+  private Schema fixedUUid = LogicalTypes.uuid().addToSchema(fixed);
+
+  @ParameterizedTest
+  @MethodSource("uuidData")
+  void uuidFixed(UUID uuid) {
+    GenericFixed value = uuidConversion.toFixed(uuid, fixedUUid, LogicalTypes.uuid());
+    UUID uuid1 = uuidConversion.fromFixed(value, fixedUUid, LogicalTypes.uuid());
+    Assertions.assertEquals(uuid, uuid1);
+  }
+
+  @ParameterizedTest
+  @MethodSource("uuidData")
+  void uuidFixedReverse(UUID uuid) {
+    Conversions.UUIDConversion bigEndianConversion = new Conversions.UUIDConversion(true);
+    Conversions.UUIDConversion littleEndianConversion = new Conversions.UUIDConversion(false);
+
+    // simulate uuid from other endian source, assume it's big endian (nevermind if
+    // current jvm is big endian, as we simulate).
+    UUID reverserUUID = new UUID(Long.reverseBytes(uuid.getMostSignificantBits()),
+        Long.reverseBytes(uuid.getLeastSignificantBits()));
```

Comment:
As a UUID is a 128-bit "label", switching between LittleEndian and BigEndian byte order also switches the order of the long values: the most significant 64 bits are in the first half for BE data, but in the last half for LE data.

Reply:
As uuid is only identifier, without any notion of ordering (you can use ordering property for treemap, but it has no functional meaning), this would have no impact.

Reply:
Yes, this point is moot: [RFC-4122 (section 4.1.2)](https://www.ietf.org/rfc/rfc4122.html#section-4.1.2) defines the byte layout.